### ISSUE TITLE
Make it easier for users to select multiple jobs/actions when submitting run request

### ISSUE THREAD

**Issue by tomodwyer** [opensafely-core/job-server#2311]
This came from a Bristol/Cambridge meeting and Simon's summary [here](https://bennettoxford.slack.com/archives/C63UXGB8E/p1667474881668509).

The longer-term fixes are in an initiative (improve project.yaml) but it would be good to discuss as a team if there is a quick fix that might be worth doing in the meantime. 

From Simon's summary:
"selecting all the actions to run is very slow and unpleasant. This is a known issue.
quickfix proposal: select actions to run via glob: https://github.com/opensafely-core/job-server/issues/1652"

**Comment by ghickman** [comment]
Could we add a way to live filter the actions displayed on the page?  I'm thinking an input box which fuzzy searches on action name, essentially what List.js does although I'm not suggesting we use that library.  The removal of actions would be a purely visual change so submitting the page wouldn't be affected by a search.

**Comment by bloodearnest** [comment]
yeah, fuzzy/prefix match and "add to job request" kinda of flow makes sense.

Also, allowing grouping or tagging of actions might be a useful way.

**Comment by eli-miriam** [comment]
We've now seen extra use cases for this arising from Em's Big Project - if they want to split their pipeline up into multiple job requests, they need to be able to somewhat programmatically select only the DB jobs, only certain subsequent analysis jobs, etc. In general, this would allow researchers to have faster feedback loops w.r.t. running jobs on the secure backend, given that they could more easily/programmatically select only parts of their pipeline.

**Comment by lucyb** [comment]
This [thread](https://bennettoxford.slack.com/archives/C01D7H9LYKB/p1732008143324159) provides an insight into some of the workarounds that are being carried out right now.
